src/assets/scripts/client/engine/TimeKeeper.js

```diff
@@ -83,35 +248,211 @@ class TimeKeeper {
      * @method reset
      */
     reset() {
-        const currentTime = this.gameTime;
+        this._accumulatedDeltaTime = 0;
+        this._elapsedFrameCount = 0;
+        this._frameDeltaTime = 0;
+        this._frameStartTimestamp = 0;
+        this._frameStep = 0;
+        this._futureTrackDeltaTimeCache = -1;
+        this._isPaused = true;
+        this._previousFrameTimestamp = 0;
+        this._startTimestamp = 0;
+        this._timescale = 1;
+    }
 
-        this.time = {};
-        this.start = currentTime;
-        this.frames = 0;
-        this.frameStartTime = currentTime;
-        this.lastFrameTime = currentTime;
-        this.frameDeltaTime = 0;
+    /**
+     * Wrapper used to get current `#_frameDeltaTime` value or zero under certain conditions
+     *
+     * When the sim is 'paused' or un-focused, we do not want `#_frameDeltaTime` to be used
+     * in position calculations. This causes movement at a time when there shouldn't be any.
+     *
+     * By supply `0` in those cases, there is effectively no 'time' difference and thus no
+     * change in position given a rate. This freezes objects at their current position until
+     * such time we begin returning `#_frameDeltaTime` again.
```

Comment:
that is something I changed, and should probably change back.  the goal there was eliminate the maths that needed to happen to calculate current time.  I'm going to add that back in, though, cause I'm not a fan of the clock running on pause.

Reply:
fixed the clock pause issue

tonight I'll look into the jump-on-unpause

Reply:
@erikquinn I'm not seeing the jump on pause with either blur or window blur?

Reply:
Now that you solved the clock issue, the jumping is resolved. Makes perfect sense that they would be linked, so that's not a surprise. I can confirm it is working as it should! :shipit: